ennMethods.rb

```diff
@@ -0,0 +1,101 @@
+# Module
+module Enumerable
+  #my_each: clone of each
+  def my_each
+    for x in self do
+      yield(x)
+    end
+    return self
+  end
+
+  #my_each_with_index: clone of each.with_index
+  def my_each_with_index
+    for i in 0...self.length do
+      yield(self[i], i)
+    end
+    return self
+  end
+
+  #my_select: clone of select
+  def my_select
+    i = 0
+    result = []
+    self.my_each do |x|
+      if yield(x)
+        result[i] = x
+        i += 1
+      end
+    end
+    return result
+  end
+
+  #my_all?: clone of all?
+  def my_all?
+    test = true
+    self.my_each do |x|
+      test = test && yield(x)
+    end
+    return test
+  end
+
+  #my_any?: clone of any?
+  def my_any?
+    test = false
+    self.my_each do |x|
+      test = test || yield(x)
+    end
+    return test
+  end
+
+  #my_none?: clone of none?
+  def my_none?
+    test = true
+    self.my_each do |x|
+      test = test || !yield(x)
+    end
+    return test
+  end
+
+  #my_count: clone of count
+  def my_count
+    counter = 0
+    self.my_each do |x|
+      counter += 1 if yield(x)
+    end
+    return counter
+  end
```

Comment:
Not working 
```
> ary.count
 => 4 
> ary.my_count
Traceback (most recent call last):
        7: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        6: from (irb):18
        5: from ennMethods.rb:62:in `my_count'
        4: from ennMethods.rb:5:in `my_each'
        3: from ennMethods.rb:5:in `each'
        2: from ennMethods.rb:6:in `block in my_each'
        1: from ennMethods.rb:63:in `block in my_count'
LocalJumpError (no block given (yield))
> ary.count(2)
 => 1 
> ary.my_count(2)
Traceback (most recent call last):
        3: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        2: from (irb):20
        1: from ennMethods.rb:60:in `my_count'
ArgumentError (wrong number of arguments (given 1, expected 0))
```

Reply:
Change made. Initially `my_count` was made to receive always a block, with a syntax similar to this:
`[1,2,12,3].my_count do |x|
  x.odd?
end`

Now it works for cases where no block is given (`[nil, true, 99].my_count`) and for cases where a pattern is given (`["area", "all", "naive"].my_count(/v/)`)